src/main/flight/position.c

```diff
@@ -107,96 +103,133 @@ void calculateEstimatedAltitude()
             pt2FilterInit(&baroDerivativeLpf, gain);
             initBaroFilter = true;
         }
-        baroAlt = baroUpsampleAltitude();
-        const float baroAltVelocityRaw = (baroAlt - lastBaroAlt) * TASK_ALTITUDE_RATE_HZ; // cm/s
+        baroAltCm = baroUpsampleAltitude();
+        const float baroAltVelocityRaw = (baroAltCm - lastBaroAltCm) * TASK_ALTITUDE_RATE_HZ; // cm/s
         baroAltVelocity = pt2FilterApply(&baroDerivativeLpf, baroAltVelocityRaw);
-        lastBaroAlt = baroAlt;
+        lastBaroAltCm = baroAltCm;
         if (baroIsCalibrated()) {
             haveBaroAlt = true;
         }
     }
 #endif
 
+    // *** Check GPS for 3D fix, set haveGpsAlt if we have a 3D fix, and GPS Trust based on hdop, or leave at default of 0.3
 #ifdef USE_GPS
     if (sensors(SENSOR_GPS) && STATE(GPS_FIX)) {
-        gpsAlt = gpsSol.llh.altCm;
-        gpsNumSat = gpsSol.numSat;
+        // Need a 3D fix, which requires min 4 sats
+        // if not, gpsAltCm remains zero, haveGpsAlt stays false, and gpsTrust is zero.
+        gpsAltCm = gpsSol.llh.altCm;
 #ifdef USE_VARIO
         gpsVertSpeed = GPS_verticalSpeedInCmS;
 #endif
-        haveGpsAlt = true;
+        haveGpsAlt = true; // remains false if no 3D fix
 
+#ifdef USE_GPS
         if (gpsSol.hdop != 0) {
             gpsTrust = 100.0 / gpsSol.hdop;
+            // *** TO DO - investigate if we should use vDOP or vACC with UBlox units; this hDOP value is actually pDOP in UBLox code !!!
         }
-        // always use at least 10% of other sources besides gps if available
+#endif
```

Comment:
Does `#ifdef USE_GPS` make sense here? Line 117 already opens up `USE_GPS` and to me it seems like it gets closed in line 139. So a `USE_GPS` inside another `USE_GPS` seems to be to no avail (from my limited perspective).

Reply:
Good catch, thanks.  
I have fixed this, and a few others, in an upcoming PR in which the 120Hz baro code is applied to GPS Rescue.
If it's OK, please approve on the understanding that I will fix this, otherwise everyone else has to re-approve.